Mathlib/Order/UpperLower/Relative.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2025 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Data.Set.Image
+import Mathlib.Data.SetLike.Basic
+import Mathlib.Order.Interval.Set.Defs
+import Mathlib.Order.SetNotation
+
+/-!
+# Properties of relative upper/lower sets
+
+This file proves results on `IsRelUpperSet` and `IsRelLowerSet`.
+-/
+
+open Set
+
+variable {α : Type*} {ι : Sort*} {κ : ι → Sort*} {s t : Set α} {a : α} {P : α → Prop}
+
+section LE
+
+variable [LE α]
+
+lemma IsRelUpperSet.prop_of_mem (hs : IsRelUpperSet s P) (h : a ∈ s) : P a := (hs h).1
+lemma IsRelLowerSet.prop_of_mem (hs : IsRelLowerSet s P) (h : a ∈ s) : P a := (hs h).1
+
+@[simp] lemma isRelUpperSet_empty : IsRelUpperSet (∅ : Set α) P := fun _ ↦ False.elim
+@[simp] lemma isRelLowerSet_empty : IsRelLowerSet (∅ : Set α) P := fun _ ↦ False.elim
+
+@[simp] lemma isRelUpperSet_self : IsRelUpperSet s s := fun _ b ↦ ⟨b, fun _ _ ↦ id⟩
```

Comment:
This is defeq abusing `Set α := α → Prop`
```suggestion
@[simp] lemma isRelUpperSet_self : IsRelUpperSet s (· ∈ s) := fun _ b ↦ ⟨b, fun _ _ ↦ id⟩
```
Same below

Reply:
OK, done.